homeassistant/components/habitica/calendar.py

```diff
@@ -0,0 +1,197 @@
+        key=HabiticaCalendar.DAILIES,
+        translation_key=HabiticaCalendar.DAILIES,
+    )
+
+    @property
+    def today(self) -> datetime:
+        """Habitica daystart."""
+        return datetime.fromisoformat(self.coordinator.data.user["lastCron"])
+
+    def calculate_end_date(self, next_recurrence: datetime) -> date:
+        """Calculate the end date for a yesterdaily."""
+        return (
+            dt_util.start_of_local_day()
+            if next_recurrence == self.today
+            else next_recurrence
+        ).date() + timedelta(days=1)
+
+    @property
+    def event(self) -> CalendarEvent | None:
+        """Return the next upcoming event."""
+
+        events = [
+            CalendarEvent(
+                start=next_recurrence.date(),
+                end=self.calculate_end_date(next_recurrence),
+                summary=task["text"],
+                description=task["notes"],
+                uid=task["id"],
+            )
+            for task in self.coordinator.data.tasks
+            if task["type"] == HabiticaTaskType.DAILY and task["everyX"]
+            if (
+                next_recurrence := self.today
+                if not task["completed"] and task["isDue"]
+                else build_rrule(task).after(self.today, inc=True)
+            )
+        ]
+        events_sorted = sorted(
+            events,
+            key=lambda event: (
+                event.start,
+                self.coordinator.data.user["tasksOrder"]["dailys"].index(event.uid),
+            ),
+        )
+
+        return next(iter(events_sorted), None)
+
+    async def async_get_events(
+        self, hass: HomeAssistant, start_date: datetime, end_date: datetime
+    ) -> list[CalendarEvent]:
+        """Return calendar events within a datetime range."""
+
+        # returns only todays and future dailies.
+        # If a daily is completed it will not be shown for today but still future recurrences
+        # If the cron hasn't run, not completed dailies are yesterdailies and displayed yesterday
+        return [
+            CalendarEvent(
+                start=start,
+                end=start + timedelta(days=1),
+                summary=task["text"],
+                description=task["notes"],
+                uid=task["id"],
+                rrule=get_recurrence_rule(task),
+            )
+            for task in self.coordinator.data.tasks
+            if task["type"] == HabiticaTaskType.DAILY and task["everyX"]
+            for recurrence in build_rrule(task).between(start_date, end_date, inc=True)
+            if (start := recurrence.date()) > self.today.date()
+            or (start == self.today.date() and not task["completed"] and task["isDue"])
+        ]
+
+    @property
+    def extra_state_attributes(self) -> dict[str, bool | None] | None:
+        """Return entity specific state attributes."""
+        return {
+            "yesterdaily": self.event.start < self.today.date() if self.event else None
+        }
```

Comment:
What does this return?

Reply:
If the task is a [Yester-Daily](https://habitica.fandom.com/wiki/Dailies#Yester-Dailies) or not.

Reply:
I'm resolving this or is there still something unclear?